### PR TITLE
Fix facet display when there is only one filter

### PR DESCRIPTION
**Pull request by PierreRambaud**
You should be able to see a facet even if you have only one filter.
See https://github.com/PrestaShop/PrestaShop/issues/9647 for more explanation.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/ps_facetedsearch/103)

### CODE REVIEW

File: src/Product/SearchProvider.php [hunk 1]
```diff
@@ -452,15 +453,29 @@ private function hideUselessFacets(array $facets)
...
+                ||
+                (
+                    /*
+                     * There is only one fitler and the
+                     * magnitude is different than the
+                     * total products
+                     */
+                    count($facet->getFilters()) === 1
+                    && $totalFacetProducts != $totalProducts
+                    && $usefulFiltersCount > 0
```
Comment (Quetzacoalt91) [inline src/Product/SearchProvider.php]: That's not the easier code to read. Maybe having the 2 comments at the same indentation would help?

Comment (PierreRambaud) [reply src/Product/SearchProvider.php]: something like.
```php
                    count($facet->getFilters()) === 1 // There is only one filter
                    && $totalFacetProducts != $totalProducts // And the magnitude is different
```

Comment (Quetzacoalt91) [reply src/Product/SearchProvider.php]:
```php
            $facet->setDisplayed(
                // There are two filters displayed
                $usefulFiltersCount > 1
                ||
                /*
                 * There is only one fitler and the
                 * magnitude is different than the
                 * total products
                 */
                (
                    count($facet->getFilters()) === 1
                    && $totalFacetProducts != $totalProducts
                    && $usefulFiltersCount > 0
                )
            );

Comment (Quetzacoalt91) [reply src/Product/SearchProvider.php]: In all cases, this test seems complex to read because of the comment in the middle of the statements  :)

Comment (PierreRambaud) [reply src/Product/SearchProvider.php]: Thanks

**Pull request merged**